src/ucs/sys/compiler.h

```diff
@@ -72,6 +72,13 @@
  */
 #define ucs_unaligned_ptr(_ptr) ((void*)(_ptr))
 
+/**
+ * suppress compilation warning on string size on gcc 8+
+ */
+#define ucs_safe_strncpy(_dst, _src, _len) \
+    memcpy(_dst, _src, ucs_min(_len, strnlen(_src, _len) + 1)); \
```

Comment:
1. pls implement it as function and not as a macro, in ucs/sys/string.h
2. what if _len == 0
3. why using strnlen, the list is on dest size not on source size

Reply:
- implemented as function.
- fixed len==0 issue
- strnlen returns length of string limited by ```len``` argument, it is safer to avoid incorrect memory access